src/extension/src/ActionHandler.py

```diff
@@ -58,27 +58,41 @@ def determine_operation(self, command):
         except KeyError as e:
             raise e
 
+    def setup(self, action, log_message):
+        self.setup_file_logger(action)
+        self.setup_telemetry()
+        self.logger.log(log_message)
+
     def setup_file_logger(self, action):
         if self.file_logger is not None or self.stdout_file_mirror is not None:
             self.logger.log_error("Log file handles from the previous operation were not closed correctly. Closing them and initializing new ones for this operation.")
-            self.close_file_logger()
+            self.tear_down()
 
         log_file_name = datetime.datetime.utcnow().strftime("%Y-%m-%d_%H-%M-%S") + "_" + str(action)
         self.file_logger = self.utility.create_log_file(self.ext_env_handler.log_folder, log_file_name)
         if self.file_logger is not None:
             self.logger.file_logger = self.file_logger
             self.stdout_file_mirror = StdOutFileMirror(self.file_logger)
 
-    def close_file_logger(self):
+    def tear_down(self):
         if self.stdout_file_mirror is not None:
             self.stdout_file_mirror.stop()
         if self.file_logger is not None:
             self.file_logger.close()
 
+    def setup_telemetry(self):
+        # check if events folder exists, if it does init telemetry, if events folder does not exist, log that telemetry is not supported by agent since events folder does not exist
+        events_folder = self.ext_env_handler.events_folder
+        if events_folder is None or not os.path.exists(events_folder):
+            err_msg = "Telemetry is not supported by Linux Guest Agent as no events folder location specified in Handler Environment. Extension will not perform the requested operation unless Linux Agent is updated to a higher version that supports telemetry. Please update Linux Agent on the VM.\n"
+            self.logger.log_error(err_msg)
+        else:
+            self.logger.log("Telemetry is supported by Linux Guest Agent. Extension will send messages to telemetry")
```

Comment:
"The minimum Azure Linux Agent version prerequisite for Linux patching was met."